File: Leetcode Challenge/March/C++/Delete and Earn.cpp
class Solution {
public:
    int deleteAndEarn(vector<int>& nums) {
        int n = 10001;
        vector<int> values(n, 0);
        for (int num : nums)
            values[num] += num;

        int take = 0, skip = 0;
        for (int i = 0; i < n; i++) {
            int takei = skip + values[i];
                int skipi = max(skip, take);
            take = takei;
            skip = skipi;
        }
        return max(take, skip);
    }
};

File: Leetcode Challenge/March/Python/DeleteandEarn.py
def deleteAndEarn(self, nums: List[int]) -> int:
        n = 10001
        values=[0]*n
        for num in nums:
            values[num] += num
        take = 0
        skip = 0
        for i in range(n):
            takei = skip + values[i]
            skipi = max(skip, take)
            take = takei
            skip = skipi
        
        return max(take, skip)
